pysat/tests/test_instrument_padding.py

```diff
@@ -0,0 +1,617 @@
+                == (self.rawInst.index[-1] + self.delta)), \
+            "failed to pad the end of the `testInst` object"
+        assert self.testInst.index.is_unique, "padded index has duplicate times"
+
+        if self.delta > dt.timedelta(seconds=0):
+            assert len(self.testInst.index) > len(self.rawInst.index), \
+                "padded instrument does not have enough data"
+        else:
+            assert len(self.testInst.index) == len(self.rawInst.index), \
+                "unpadded instrument has extra or is missing data"
+        return
+
+    @pytest.mark.parametrize("dmin,tind,ncycle", [
+        (5, 1, 0), (5, 1, 1), (5, 1, 2), (5, 2, -1), (5, 10, -2)])
+    def test_fname_data_padding(self, dmin, tind, ncycle):
+        """Test data padding load by filename."""
+
+        # Load the test data with padding
+        self.testInst.load(fname=self.testInst.files[tind], verifyPad=True)
+
+        rind = tind + ncycle
+        if ncycle > 0:
+            while ncycle > 1:
+                self.testInst.next()
+                ncycle -= 1
+            self.testInst.next(verifyPad=True)
+        elif ncycle < 0:
+            while ncycle < 1:
+                self.testInst.prev()
+                ncycle += 1
+            self.testInst.prev(verifyPad=True)
+
+        # Load the comparison file without padding and set the padding time
+        self.rawInst.load(fname=self.testInst.files[rind])
+        self.delta = dt.timedelta(minutes=dmin)
+
+        # Evaluate the test results
+        self.eval_index_start_end()
+        return
+
+    def test_fname_data_padding_jump(self):
+        """Test data padding by filename after loading non-consecutive file."""
+
+        self.testInst.load(fname=self.testInst.files[1], verifyPad=True)
+        self.testInst.load(fname=self.testInst.files[10], verifyPad=True)
+        self.rawInst.load(fname=self.testInst.files[10])
+        self.delta = dt.timedelta(minutes=5)
+        self.eval_index_start_end()
+        return
+
+    def test_fname_data_padding_uniqueness(self):
+        """Ensure uniqueness data padding when loading by file."""
+
+        self.testInst.load(fname=self.testInst.files[1], verifyPad=True)
+        assert self.testInst.index.is_unique
+        return
+
+    def test_fname_data_padding_all_samples_present(self):
+        """Ensure all samples present when padding and loading by file."""
+
+        self.testInst.load(fname=self.testInst.files[1], verifyPad=True)
+        self.delta = pds.date_range(self.testInst.index[0],
+                                    self.testInst.index[-1], freq='S')
+        assert np.all(self.testInst.index == self.delta)
+        return
+
+    def test_fname_data_padding_removal(self):
+        """Ensure padded samples nominally dropped, loading by file."""
+
+        self.testInst.load(fname=self.testInst.files[1])
+        self.rawInst.load(fname=self.testInst.files[1])
+        self.eval_index_start_end()
+        return
```

Comment:
Not sure what this test is doing; since no padding is implemented, it can't be removed.

Reply:
testInst has padding enabled at instantiation. This test is verifying that the padded samples are automatically removed before finishing the load process by comparison to a non-padded Instrument load. the `verifyPad=True` option disables the final padded data filter in the other tests.